kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/validation/asset/DefaultGenericKieValidator.java

```
@@ -0,0 +1,60 @@
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.workbench.common.services.backend.validation.asset;
+
+import java.io.InputStream;
+import java.util.List;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.guvnor.common.services.backend.validation.GenericValidator;
+import org.guvnor.common.services.shared.validation.model.ValidationMessage;
+import org.kie.workbench.common.services.shared.project.KieProjectService;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.io.IOService;
+import org.uberfire.java.nio.file.DirectoryStream;
+
+/**
+ * Validator capable of validating generic Kie assets (i.e those that are handled by KieBuilder)
+ */
+public class DefaultGenericKieValidator
+        implements GenericValidator {
+
+    private IOService ioService;
+
+    private KieProjectService projectService;
+
+    public DefaultGenericKieValidator() {
+    }
+
+    @Inject
+    public DefaultGenericKieValidator( final @Named( "ioStrategy" ) IOService ioService,
+                                       final KieProjectService projectService ) {
+        this.ioService = ioService;
+        this.projectService = projectService;
+    }
+
+    public List<ValidationMessage> validate( final Path resourcePath,
+                                             final InputStream resource,
+                                             final DirectoryStream.Filter<org.uberfire.java.nio.file.Path>... supportingFileFilters ) {
+
+        return new Validator( new ValidatorFileSystemProvider( resourcePath,
```

That is what this code does too. We check the project exists before running the the validation in Validator.validate().
But you are right. The goal here is make reusable bits, so ValidatorFileSystemProvider should not be usable in a way that can cause a NPE.
And your comments made me think we should just fail fast if the asset is not in a project.